modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_impl.ftl

```diff
@@ -1220,10 +1218,6 @@ public abstract class Base${schemaName}ResourceImpl
 	}
 
 	public void setContextHttpServletRequest(HttpServletRequest contextHttpServletRequest) {
-		if ((contextHttpServletRequest != null) && (contextHttpServletRequest.getAttribute(WebKeys.CTX) == null)) {
```

Comment:
I think this is not backwards compatible. 

REST Builder is globally published for all the versions of DXP. That means we have to keep backwards compatibility forever. 

If we remove this and a customer in a previous version reruns buildRest this call will be removed from their code and they will not have the latest context container request filter. 

@ccorreagg what do you think? Options that come to my mind:
* Use `compatibilityVersion` for this...
* Leave it as it is and just update `resource_factory_impl.ftl`.

Reply:
it's ok if we revert this and leave it back on the setter. 

I may be wrong, anyways, this change was added to the baseResourceImpl [in october 2023](https://github.com/liferay/liferay-portal/commit/f3cd9827aaf6febaddfb8c5d803e6859a45c9d55), so the range of affected versions spans for 6 months. 

Not sure about whether past versions of `ContextContainerRequestFilter.java` used the setter or not, however it looks like the resource impl setter was not being used to set this value before october 2023. So, it seems the FactoryImpl is the only place where this couuld be a concern, however as this commit sets the CTX in the factory, right before callin the setter, things should be ok.

Anyways you guys have the last word on this, I'm fine either way. We'll have to tell Brian about why we do need to duplicate the logic as he was concerned with the [old solution I provided](https://github.com/brianchandotcom/liferay-portal/pull/148917#issuecomment-2041706689)

Reply:
I am checking the usages of the `setContextHttpServletRequest` method of the `Base*ResourceImpl` class and the factory is not the only one that uses it.

I see it is from the `Query.java` and `Mutation.java` classes too, so I guess we should apply the same pattern to those classes if we want to support also from the GraphQL side and not only from the REST API side.

I am reviewing also the changes that Shuyang performed and I am not sure that he didn't break anything. For example, he replaced this code of the `SitePageResourceImpl`

![image](https://github.com/liferay-headless/liferay-portal/assets/32699538/ad7aee28-c7c8-4500-ae91-c14c507c398e)

with the code in the setter. but the setter is not being executed from the REST side when performing a request, so, are we losing that attribute since then?